src/events/issues/mod.rs

```diff
@@ -0,0 +1,190 @@
+                member.permissions.map_or(false, |member_perms| {
+                    member_perms.contains(Permissions::MANAGE_MESSAGES)
+                })
+            });
+
+            // Users who do not own the message or have permissions cannot execute the interactions.
+            if !(press.user.id == message.author.id || has_perms) {
+                interaction_err(
+                    ctx,
+                    &press,
+                    "Unable to use interaction because you are missing `MANAGE_MESSAGES`.",
+                )
+                .await;
+
+                continue;
+            }
+
+            match Kind::from_id(&press.data.custom_id, &ctx_id) {
+                Some(Kind::Delete) => {
+                    let _ = press
+                        .create_response(ctx, CreateInteractionResponse::Acknowledge)
+                        .await;
+                    if let Ok(ref msg) = msg_result {
+                        let _ = msg.delete(ctx).await;
+                    }
+                    msg_deleted = true;
+                }
+                Some(Kind::HideBody) => {
+                    if !body_hid {
+                        let mut hid_body_embeds: Vec<CreateEmbed> = Vec::new();
+                        if let Ok(ref msg) = msg_result {
+                            for mut embed in msg.embeds.clone() {
+                                embed.description = None;
+                                let embed: CreateEmbed = embed.clone().into();
+                                hid_body_embeds.push(embed);
+                            }
+                        }
+
+                        let _ = press
+                            .create_response(
+                                ctx,
+                                serenity::CreateInteractionResponse::UpdateMessage(
+                                    serenity::CreateInteractionResponseMessage::new()
+                                        .embeds(hid_body_embeds)
+                                        .components(vec![remove.clone()]),
+                                ),
+                            )
+                            .await;
+                    }
+                    body_hid = true;
+                }
+                None => {}
+            }
+        }
+
+        // Triggers on timeout.
+        if !msg_deleted {
+            if let Ok(mut msg) = msg_result {
+                let _ = msg
+                    .edit(ctx, serenity::EditMessage::default().components(vec![]))
+                    .await;
+            }
+        }
+    }
+}
+
+async fn issue_embeds(data: &Data, message: &Message) -> Option<Vec<CreateEmbed>> {
+    let mut embeds: Vec<CreateEmbed> = vec![];
+    let client = octocrab::instance();
+    let ratelimit = client.ratelimit();
+
+    // TODO: stop compiling this every time.
+    let regex = Regex::new(r" ?([a-zA-Z0-9-_.]+)?#([0-9]+) ?").expect("Expected numbers regex");
```

Comment:
Would be a good time to make this static

Reply:
I opened #35 about it, theres 4 usages of Regex::new() off the top of my head and none of them are static, its not restricted to just this area, i just forgot to remove this TODO after realizing that its not isolated.